packages/ui-components/src/components/windowedlist.ts

```diff
@@ -1182,18 +1278,58 @@ export class WindowedList<
     }
   }
 
-  protected _viewModel: T;
-  private _innerElement: HTMLDivElement;
+  /**
+   * Render virtual scrollbar.
+   */
+  private _renderScrollbar(): void {
```

Comment:
Note for future: there does not have a guard preventing regeneration when nothing changed, this means it is recreated on every cursor movement in resize.

Reply:
I started to tackle this problem by creating a scroll state variable that holds the list of `selected` items and the index of the `active` item. But the trouble is that part of the "state" of the scrollbar is out of our hands because the renderer that generates each item in the scrollbar might change arbitrarily, for example it could include a timestamp in it or something else that is nondeterministic.

So I think the only way to approach this would be to use a virtual DOM (whether React or vdom, etc.) and let that decision be made by the tree-diffing algorithm.